File: CRYPTOGRAPH/chall/Golang-Ristek Comvis/chall.go
package main

import (
	"bufio"
	"crypto/aes"
	"crypto/cipher"
	"encoding/hex"
	"fmt"
	"os"
	"strings"
)

const (
	secretKey = "PEKANRIZZTEK2024"
	flagFile  = "flag.txt" 
)

func pad(data []byte, blockSize int) []byte {
	padding := blockSize - len(data)%blockSize
	padtext := make([]byte, padding)
	for i := range padtext {
		padtext[i] = byte(padding)
	}
	return append(data, padtext...)
}

func unpad(data []byte) []byte {
	length := len(data)
	unpadding := int(data[length-1])
	return data[:(length - unpadding)]
}

func decryptECB(block cipher.Block, data []byte) []byte {
	blockSize := block.BlockSize()
	decrypted := make([]byte, len(data))
	
	for i := 0; i < len(data); i += blockSize {
		block.Decrypt(decrypted[i:i+blockSize], data[i:i+blockSize])
	}
	return unpad(decrypted)
}

func decrypt(ciphertext string) (string, error) {
	decoded, err := hex.DecodeString(ciphertext)
	if err != nil {
		return "", err
	}
	
	block, err := aes.NewCipher([]byte(secretKey))
	if err != nil {
		return "", err
	}
	
	decrypted := decryptECB(block, decoded)
	return string(decrypted), nil
}

func readFlag() string {
	data, err := os.ReadFile(flagFile)
	if err != nil {
		return "Error: Could not read flag file!"
	}
	return string(data)
}

func menu() int {
	fmt.Println("1. Decrypt Message")
	fmt.Println("2. Get Flag")
	fmt.Println("3. Exit")
	fmt.Print(">> ")

	reader := bufio.NewReader(os.Stdin)
	input, err := reader.ReadString('\n')
	if err != nil {
		return 0
	}
	
	input = strings.TrimSpace(input)
	var choice int
	_, err = fmt.Sscanf(input, "%d", &choice)
	if err != nil {
		return 0
	}
	
	return choice
}

func main() {
	fmt.Println("Welcome to ComVis Selection :>")
	reader := bufio.NewReader(os.Stdin)
	
	for {
		choice := menu()
		switch choice {
		case 1:
			fmt.Print("Enter ciphertext (hex): ")
			ciphertext, _ := reader.ReadString('\n')
			ciphertext = strings.TrimSpace(ciphertext)
			
			decrypted, err := decrypt(ciphertext)
			if err != nil {
				fmt.Println("Invalid ciphertext.")
				continue
			}
			
			fmt.Println("Decrypted:", decrypted)
			
		case 2:
			fmt.Print("Enter the encrypted messages (hex): ")
			ciphertext, _ := reader.ReadString('\n')
			ciphertext = strings.TrimSpace(ciphertext)
			
			decrypted, err := decrypt(ciphertext)
			if err != nil {
				fmt.Println("Invalid ciphertext.")
				continue
			}
			
			if strings.Contains(decrypted, "NETSOS") {
				fmt.Println("Success! Enjoy your flag!")
				flag := readFlag()
				fmt.Println(flag)
				os.Exit(0)
			} else {
				fmt.Println("Invalid message!")
			}
			
		case 3:
			os.Exit(0)
			
		default:
			fmt.Println("Invalid choice!")
		}
	}
}

File: CRYPTOGRAPH/chall/CBC with P-Gexxtik/chall.py
from binascii import hexlify, unhexlify
from os import urandom
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
from random import randint

key = urandom(AES.key_size[0])
secret = urandom(randint(AES.block_size, 4 * AES.block_size))

def xor(a, b):
    return bytes([a[i] ^ b[i] for i in range(len(a))])

def encrypt(msg):
    pt = pad(msg, AES.block_size)
    iv = urandom(AES.block_size)
    ct = iv[:]
    for i in range(0, len(pt), AES.block_size):
        cipher = AES.new(key, AES.MODE_CBC, iv)
        ct += cipher.encrypt(pt[i:i+AES.block_size])
        iv = xor(ct[-AES.block_size:], pt[i:i + AES.block_size])
    return ct

def decrypt(ct):
    iv = ct[:AES.block_size]
    pt = b''
    for i in range(AES.block_size, len(ct), AES.block_size):
        cipher = AES.new(key, AES.MODE_CBC, iv)
        pt += cipher.decrypt(ct[i:i+AES.block_size])
        iv = xor(ct[i:i + AES.block_size], pt[-AES.block_size:])
    return unpad(pt, AES.block_size)


rnd = urandom(AES.block_size)
assert decrypt(encrypt(rnd)) == rnd

def menu():
    print('1. Get encrypted secret')
    print('2. Decrypt a message')
    print('3. Get flag')
    print('4. Exit')

while True:
    try:
        menu()
        choice = input('> ')
        if choice == '1':
            res = hexlify(encrypt(secret)).decode()
            print(res)
        elif (choice == '2'):
            ct = unhexlify(input('> '))
            decrypt(ct)
        elif (choice == '3'):
            answer = unhexlify(input('> ').strip())
            if answer == secret:
                print(open('flag.txt', 'rb').read())
            break
        elif (choice == '4'):
            print('Bye.')
            break
        else:
            print('Invalid input!')
    except:
        print('Something went wrong.')